File: zcgl/src/main/java/com/bjprd/zcgl/main/MainActivity.java
package com.bjprd.zcgl.main;

import com.bjprd.zcgl.R;
import com.bjprd.zcgl.base.AppBarViewModel;
import com.bjprd.zcgl.base.BaseActivity;
import com.bjprd.zcgl.databinding.ActivityMainBinding;

/**
 * Created by 王少岩 on 2016/10/19.
 */

public class MainActivity extends BaseActivity<ActivityMainBinding> {

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void setViewModel() {
        getMBinding().setViewModel(new MainViewModel());
        getMBinding().setAppbar(new AppBarViewModel("固定资产管理系统", false));
    }

    @Override
    protected void init() {
        getMBinding().drawerLayout.findViewById(R.id.tv_home).setOnClickListener(v -> getMBinding().drawerLayout.closeDrawers());
    }
}

File: Zcgl/src/main/java/com/bjprd/zcgl/main/MainActivity.kt
package com.bjprd.zcgl.main

import com.bjprd.zcgl.R
import com.bjprd.zcgl.base.AppBarViewModel
import com.bjprd.zcgl.base.BaseActivity
import com.bjprd.zcgl.databinding.ActivityMainBinding

/**
 * Created by 王少岩 on 2016/10/19.
 */

class MainActivity : BaseActivity<ActivityMainBinding>() {

    override val layoutId: Int
        get() = R.layout.activity_main

    override fun setViewModel() {
        mBinding!!.viewModel = MainViewModel()
        mBinding!!.appbar = AppBarViewModel("固定资产管理系统", false)
    }

    override fun init() {
        mBinding!!.drawerLayout.findViewById(R.id.tv_home).setOnClickListener { v -> mBinding!!.drawerLayout.closeDrawers() }
    }
}
